node/src/components/block_synchronizer/global_state_synchronizer.rs

```diff
@@ -32,21 +32,44 @@ const COMPONENT_NAME: &str = "global_state_synchronizer";
 
 #[derive(Debug, Clone, Error)]
 pub(crate) enum Error {
-    #[error(transparent)]
-    TrieAccumulator(TrieAccumulatorError),
-    #[error("ContractRuntime failed to put a trie into global state: {0}")]
-    PutTrie(engine_state::Error),
+    #[error("trie accumulator encountered an error while fetching a trie; unreliable peers {0:?}")]
```

Comment:
I think we should avoid mixing debug output into display output.  There's `utils::DisplayIter` which could probably help here.

Same comment for the `PutTrie` variant below.

Reply:
Fixed in https://github.com/casper-network/casper-node/pull/3594/commits/18b802deab67b7404bb7570a085b695e73a9b8f1